app/controllers/general.php

```diff
@@ -240,33 +240,22 @@
             $role = Auth::USER_ROLE_APP;
             $scopes = \array_merge($roles[$role]['scopes'], $key->getAttribute('scopes', []));
 
+            Authorization::setRole('role:'.Auth::USER_ROLE_APP);
             Authorization::setDefaultStatus(false);  // Cancel security segmentation for API keys.
         }
     }
 
-    if ($user->getId()) {
-        Authorization::setRole('user:'.$user->getId());
+    foreach (Auth::getRoles($user) as $authRole) {
+        Authorization::setRole($authRole);
```

Comment:
Is there any effect of iterating over this list and using `setRole` over and over again ? 

Isn't this the same as doing `Authorization::setRole(end(Auth::getRoles($user)))`

Reply:
I removed code later that would iterate anyway, I just optimized it here 👍🏻